Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -314,23 +324,46 @@ void OpenALStream::SoundLoop()
     // many silence samples.  These do not need to be timestretched.
     if (rate > 0.10)
     {
-      soundTouch.setTempo(rate);
-      if (rate > 10)
+      if (use_timestretching)
+      {
+        soundTouch.setTempo(rate);
+      }
+      else
+      {
+        alSourcef(ui_source, AL_PITCH, static_cast<ALfloat>(rate));
```

Comment:
Now I understand! Just run some tests to compare both Dolphin's native frequency-shifting (cosider_framelimite = true), OpenAL Soft's and X-Fi's (consider_framelimit = false).

Dolphin's is OK, but probably was made to be fast and easy on the CPU. It doesn't shift the frequency very much (which is good) but causes "pockets" of no sound (stuttering?) (which is bad).
OpenAL Soft's and X-Fi's shifts the frequency a lot (which is bad) but the sound is smooth as butter.

I can record a video comparing all of them if you want.

I noticed no increase in latency, and the good thing about using OpenAL to frequency-shift is that we can update the pitch/playback-speed every time m_mixer->GetCurrentSpeed() changes, not only after a buffer is played. I admit that it won't change much as each buffer is really small (10ms or lower)...